lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -51,14 +51,7 @@ sub set_engine_exec ($path) {
 
 sub _save_vars ($pooldir, $vars) {
     die 'cannot get environment variables!\n' unless $vars;
-    my $fn = $pooldir . '/vars.json';
-    unlink "$pooldir/vars.json" if -e "$pooldir/vars.json";
-    open(my $fd, '>', $fn) or die "can not write vars.json: $!\n";
-    fcntl($fd, F_SETLKW, pack('ssqql', F_WRLCK, 0, 0, 0, $$)) or die "cannot lock vars.json: $!\n";
-    truncate($fd, 0) or die "cannot truncate vars.json: $!\n";
-
-    print $fd Cpanel::JSON::XS->new->pretty(1)->encode(\%$vars);
-    close($fd);
+    path($pooldir, 'vars.json')->spew(Cpanel::JSON::XS->new->pretty(1)->encode(\%$vars));
```

Comment:
The previous code tried to handle the case when this function is executed concurrently. Is this also handled by `spew`? (This is of course irrelevant when the function is not actually invoked concurrently.)

But to be fair, I guess the previous code didn't handle this in a good way, too. It actually tried to unlink the file before acquiring the lock and I am not sure whether that's a good idea.

Reply:
I was wondering about that, too.
But at least the code was assuring that the file contained valid JSON, I think.
`spew` won't guarantee that.
But i'm wondering as well when our code would concurrently try to write this file.
To be safe, maybe the easiest thing here would be to first write to `"$pooldir/vars.json.$$" and then `mv` it to its target.

Reply:
Can you elaborate under which circumstances we would need to first write to a temp file and then rename to the target?

Reply:
> > … when this function is executed concurrently

but when would this code be executed concurrently for one individual worker instance?

Reply:
Looks like this function is only executed once and that's before creating the isotovideo process. So I guess it is ok not to worry about concurrent execution then.

Reply:
Yes, one of those